iree/hal/vulkan/native_event.cc

```diff
@@ -22,7 +22,9 @@ typedef struct iree_hal_vulkan_native_event_t {
   VkEvent handle;
 } iree_hal_vulkan_native_event_t;
 
+namespace {
 extern const iree_hal_event_vtable_t iree_hal_vulkan_native_event_vtable;
```

Comment:
I'm not sure what this is trying to do (extern in an anonymous namespace)

Reply:
In C++ the `extern` is required to forward declare the value that's defined below and the anonymous namespace keeps the whole thing local. Issue is that in C the same identifier used within a compilation unit refers to the same instance (so two `static int foo` will reference the same memory, only one of which may contain a definition), while in C++ they don't and so the extern is effectively telling the compiler not to unique it. It's ... not a stellar feature of the language :P